deployments-core/src/main/java/org/ccci/deployment/util/PageMatcher.java

```diff
@@ -76,29 +75,24 @@ public Matcher pingUntilPageMatches(String uri, String regularExpression, String
             try
             {
                 HttpResponse response = httpclient.execute(request);
-                try
+                final String content = EntityUtils.toString(response.getEntity());
+
+                StatusLine statusLine = response.getStatusLine();
+                if (statusLine.getStatusCode() == 200)
                 {
-                    StatusLine statusLine = response.getStatusLine();
-                    if (statusLine.getStatusCode() == 200)
-                    {
-                        return matchPage(response, successPattern, regularExpression, pageName);
-                    }
-                    else if (statusLine.getStatusCode() == 404)
-                        log.debug("received 404; continuing to try to connect");
-                    else if(statusLine.getStatusCode() == 400 && statusLine.getReasonPhrase().contains("No Host matches server name"))
-                    {
-                        // Sometimes Tomcat responds with a 400 'No Host matches server name <hostname>' when
-                        // the webapp is still loading.
-                        log.debug("received 400 'No Host matches'; continuing to try to connect");
-                    }
-                    else
-                        throw new RuntimeException(pageName + " not available; status line: " + statusLine);
+                    return matchPage(successPattern, regularExpression, pageName, content);
                 }
-                finally
-                //free up http connection
+                else if (statusLine.getStatusCode() == 404)
+                    log.debug("received 404; continuing to try to connect");
+                else if(statusLine.getStatusCode() == 400 && statusLine.getReasonPhrase().contains("No Host matches server name"))
                 {
-                    response.getEntity().consumeContent();
```

Comment:
I take it `response.getEntity().consumeContent()` is no longer required due to the introduction of `final String content = EntityUtils.toString(response.getEntity())` (https://github.com/CruGlobal/deployments/pull/6/files#diff-96bc565aa35a123a166dd274313949c1R78).

Was that required as a result of the upgrade or simply a refactoring improvement?

Reply:
Right.

Result of the upgrade.